### DISCUSSION TITLE
How could I do some DB initialization if I still want to use existing Poolable?

### DISCUSSION THREAD

**Discussion by Magicloud** [rwf2/Rocket#1749] [Questions]
The problem is about Sqlite. There are some pragmas I want to set. But some of them are not persistent, I need to set them every time I start the Rocket application.

How could I do it in minimal code?

**Comment by SergioBenitez** [comment]
Many of these options can be set via the URL. What are you trying to set?

Unfortunately, not all drivers allow configuring connections as they're placed in the pool, so exposing an interface to do so is a bit difficult.

**Reply by Magicloud** [reply]
That'd be "PRAGMA journal_mode = WAL", "PRAGMA synchronous = NORMAL", "PRAGMA foreign_keys = ON", "PRAGMA busy_timeout = 15". As you may know, those are executed as standard SQL query.

**Reply by SergioBenitez** [reply]
Rocket already executes the following for diesel/SQLite (note that `busy_timeout` is in milliseconds):

```
PRAGMA journal_mode = WAL;
PRAGMA busy_timeout = 1000;
PRAGMA foreign_keys = ON;
```

This isn't documented, it seems. I'd be happy to accept a PR that documents this. So if you're using diesel/sqlite, the only thing that you'd be changing is `busy_timeout` to something _very_ low and `synchronous` to `NORMAL` from `FULL`.

`r2d2` _does_ provide a generic hook for customizing connections, and I'd accept a PR to enable passing on a customizer to `r2d2` in `rocket_sync_db_pools`. 

Nevertheless, we'll advocate for using `rocket_db_pools`, an `async` variant, in the next RC release in place of `rocket_sync_db_pools`. `deadpool`, one of the supported pools, provides no such analog, unfortunately, so we'd need to first have `deadpool` support this.

**Reply by Magicloud** [reply]
I am not sure I followed "already executes". By default, Rocket + Diesel usage does not have the WAL file generated. Did you mean directly using SQLite, not via Diesel?

**Reply by SergioBenitez** [reply]
No, I mean exactly that Rocket executes that SQL for every connection in the Diesel/SQLite database pool:

https://github.com/SergioBenitez/Rocket/blob/7761911847d88a0669e86d503215b3759b27d667/contrib/sync_db_pools/lib/src/poolable.rs#L116-L138

**Reply by Magicloud** [reply]
Took some time trying to see why mine app does not look like this. Then I noticed this is in 0.5 rc ....
